fdbrpc/FlowTransport.actor.cpp

```diff
@@ -1446,18 +1447,11 @@ bool FlowTransport::incompatibleOutgoingConnectionsPresent() {
 }
 
 void FlowTransport::createInstance(bool isClient, uint64_t transportId) {
-	g_network->setGlobal(INetwork::enFailureMonitor, (flowGlobalType) new SimpleFailureMonitor());
-	g_network->setGlobal(INetwork::enClientFailureMonitor, isClient ? (flowGlobalType)1 : nullptr);
 	g_network->setGlobal(INetwork::enFlowTransport, (flowGlobalType) new FlowTransport(transportId));
 	g_network->setGlobal(INetwork::enNetworkAddressFunc, (flowGlobalType) &FlowTransport::getGlobalLocalAddress);
 	g_network->setGlobal(INetwork::enNetworkAddressesFunc, (flowGlobalType) &FlowTransport::getGlobalLocalAddresses);
-
-	// Mark ourselves as avaiable in FailureMonitor
-	const auto& localAddresses = FlowTransport::transport().getLocalAddresses();
-	IFailureMonitor::failureMonitor().setStatus(localAddresses.address, FailureStatus(false));
-	if (localAddresses.secondaryAddress.present()) {
-		IFailureMonitor::failureMonitor().setStatus(localAddresses.secondaryAddress.get(), FailureStatus(false));
-	}
+	g_network->setGlobal(INetwork::enFailureMonitor, (flowGlobalType) new SimpleFailureMonitor());
```

Comment:
Since we are calling the `SimpleFailureMonitor` constructor here, will this still be calling `setStatus` from a different thread?

Reply:
No, the reason I moved that code inside SimpleFailureMonitor since it can change the status internally and avoid calling `setStatus()`. Not ideal, but since this a special initialization case it should be fine and we would almost never care about log for this. I think the only way to force running `setStatus()` in network thread at that stage, would be to start a ACTOR?

Reply:
Ah I see. An actor could do it, as could potentially moving the initialization to somewhere later on the run loop, but your approach seems fine too. It seems ok that we omit the trace event here, especially given that the suppression means we aren't guaranteed to see a trace event for every case anyway.